### PR TITLE
add logic to withdraw and validate selected current vest and estimate gas fee

### PR DESCRIPTION
**Pull request by mwiniarsk**

### CODE REVIEW

File: src/pages/Staking/VestsList/WithdrawVest/WithdrawVest.hooks.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { FOUR_WEEKS } from '../../../../constants';
+import {
+  currentBlockSelector,
+  stakingContractSelector,
+} from '../../../../store/app/app.selectors';
+import { selectedVestSelector } from '../../../../store/staking/staking.selectors';
+
+export const useGetUnlockedVesting = () => {
```
Comment (meho99) [inline src/pages/Staking/VestsList/WithdrawVest/WithdrawVest.hooks.ts:11]: @mwiniarsk @michalsidzej 
Guys let's discuss if we want to move this to saga that would be triggered on each block.

Comment (michalsidzej) [reply src/pages/Staking/VestsList/WithdrawVest/WithdrawVest.hooks.ts:11]: I'd leave it for now, just make sure we refactor it after the app is ready

Comment (mwiniarsk) [reply src/pages/Staking/VestsList/WithdrawVest/WithdrawVest.hooks.ts:11]: hmm, wouldn't it be updated even on each block by useEffect -  `[selectedVestData, staking, currentBlockNumber]` when modal is active? but I agree with @michalsidzej if we plan to refactor code after some progress we can stick to this implementation. Also, we can discuss similar cases, if we want to use saga or not.

Comment (meho99) [reply src/pages/Staking/VestsList/WithdrawVest/WithdrawVest.hooks.ts:11]: ok, makes sense. Let's plan the refactor when all the features are ready

**Review by meho99 — APPROVED** [review]
LGTM

**Pull request merged**